tools/bigquery-zos-mainframe-connector/README.md

```diff
@@ -1,326 +1,172 @@
+
+* gszutil: provides gsutil and bq equivalent utilities for IBM mainframes running z/OS.
 
 
 ## Usage
+### Using BQSH from JCL
 
-Users can make multiple calls in a single step by entering commands on separate lines or delimited by semicolon (`;`).
+All usage is driven by JCL and divided into Job Steps. Input datasets are provided by a DD statement and commands are provided by STDIN stream input. STDIN is read from all lines between "//STDIN DD *" and "/*" in each step.
 
-Typically a JES job will contain multiple steps, each step executing the `BQSH` procedure with one or more `gsutil` or `bq` commands.
+Users can execute multiple commands in a single step by entering commands on separate lines or delimited by semicolon.
 
+Typically a JES job will contain multiple steps, each step executing the BQSH procedure with one or more gsutil or bq commands.
 
-### Example JCL
 
+### DD Names (Dataset Definition)
+
+There are a fixed set of DD names that are expected by the utility. Whether each one is required or not depends on the command being called. All MVS Datasets referenced by a DD must used Fixed Block (FB) record format.
+
+#### KEYFILE
+MVS Dataset containing a Google Cloud IAM service account JSON keyfile.
+
+#### INFILE
+MVS Dataset containing COBOL dataset to be uploaded to Google Cloud Storage.
+
+#### COPYBOOK
+MVS Dataset containing a COBOL CopyBook for the dataset referenced by the INFILE DD.
+
+#### STDIN
+Stream input used to provide shell commands.
+
+#### QUERY
+MVS Dataset containing a BigQuery Standard SQL Query.
+
+
+### Example JCL
 This example has 3 steps:
-1. Upload COBOL dataset as ORC to a bucket
-2. Load ORC file to BigQuery table
-3. Query BigQuery table
+* Upload COBOL dataset as ORC to a bucket
+* Load ORC file to BigQuery table
+* Query BigQuery table
 
 ```
 //STEP01 EXEC BQSH
 //INFILE DD DSN=<HLQ>.DATA.FILENAME,DISP=SHR
 //COPYBOOK DD DISP=SHR,DSN=<HLQ>.COPYBOOK.FILENAME
 //STDIN DD *
-gsutil cp --replace gs://bucket/tablename.orc
+BUCKET=my-long-bucket-name-1234
+gsutil cp --replace gs://$BUCKET/tablename.orc
 /*
 //COND02 IF (STEP01.RC = 0) THEN
 //STEP02 EXEC BQSH
 //STDIN DD *
-bq load --project_id=myproject \
+PROJECT=my-long-project-name-1234
+bq load --project_id=$PROJECT \
   myproject:DATASET.TABLE \
   gs://bucket/tablename.orc/*
 /*
 //COND03 IF (STEP02.RC = 0) THEN
 //STEP03 EXEC BQSH
-//QUERY DD DSN=<HLQ>.QUERY.FILENAME,DISP=SHR
+//QUERY DD DSN=<HLQ>.QUERY.FILENAME,DISP=SHR 
 //STDIN DD *
-bq query --project_id=myproject \
+PROJECT=my-long-project-name-1234
+bq query --project_id=$PROJECT \
   myproject:DATASET.TABLE \
   gs://bucket/tablename.orc/*
 /*
 ```
 
+### Enabling Debug Logging
+Debug logging can be enabled by editing the BQSH Procedure and setting the environment variable BQSH_ROOT_LOGGER=DEBUG \
+Debug logging can be disabled by commenting out the line or setting the variable to anything other than DEBUG.
 
-### Subcommand Usage
-
-#### gsutil
+## Using with Google Cloud Storage
 
```

Comment:
recommend adding the gszutil command here and stating the difference between gsutil and gszutil as whether the grpc service is optional or not. (gszutil command is streamlined because it assumes the grpc server is used and this is the command expected to be used in prod for most cases)